src/test/perl/lock.t

```diff
@@ -38,4 +46,90 @@ ok($lock2->set_lock(), "lock2 set when lock1 unlocked");
 ok($lock2->is_set(), "lock2 locked on request");
 ok(!$lock1->is_set(), "lock1 unlocked when lock2 unlocked");
 
+
+=head1 oldtsyle compatibility tests
+
+=cut
+
+
+sub makefile
+{
+    my $fn = shift;
+    open(FH, ">$fn");
+    my $txt = shift;
+    print FH (defined($txt) ? $txt : "ok");
+    close(FH);
+}
+
+sub readfile
+{
+    open(FH, shift);
+    my $txt = join('', <FH>);
+    close(FH);
+    return $txt;
+}
+
+# Make oldstyle non-stale lock
+$lock1 = undef;
+unlink(LOCK_TEST);
+
+my $pid = 0;
+ok(kill(0, $pid), "PID $pid exists");
+makefile(LOCK_TEST, "$pid");
+
+$lock1 = CAF::Lock->new(LOCK_TEST, log => $obj);
+ok($lock1->_is_locked_oldstyle(),
+   "Presence of non-empty lockfile is an old-style lock (by other process) and pid exists");
+ok(! $lock1->_is_locked_oldstyle(FORCE_ALWAYS),
+   "Presence of non-empty lockfile is not an old-style lock (by other process) if FORCE_ALWAYS is used and pid exists");
+ok($lock1->_is_locked_oldstyle(FORCE_IF_STALE),
+   "Presence of non-empty lockfile is an old-style lock if FORCE_IF_STALE is used and pid exists");
+
+# Make oldstyle stale lock
+$lock1 = undef;
+unlink(LOCK_TEST);
+
+# will be a while till this is becomes a possible PID;
+# should also not bee too  big so perl still thinks this is an interger
+$pid = 2**31;
+like("$pid", qr{^\d+$}, "too high pid $pid should still be integer");
+ok(! kill(0, "$pid"), "no such PID $pid");
+makefile(LOCK_TEST, "$pid");
+
+$lock1 = CAF::Lock->new(LOCK_TEST, log => $obj);
+ok($lock1->_is_locked_oldstyle(),
+   "Presence of non-empty lockfile is an old-style lock (by other process) and stale pid");
+ok(! $lock1->_is_locked_oldstyle(FORCE_ALWAYS),
+   "Presence of non-empty lockfile is not an old-style lock (by other process) if FORCE_ALWAYS is used and stale pid");
+ok(! $lock1->_is_locked_oldstyle(FORCE_IF_STALE),
+   "Presence of non-empty lockfile is not an old-style lock if FORCE_IF_STALE is used and stale pid");
+
+# Take a FORCE_IF_STALE lock
```

Comment:
I don't have the feeling that `FORCE_IF_STALE` is a type of lock... what aobut "Test FORCE_IF_STALE flag used to take a lock when a an old-style lock file is present but the pid which created it no longer exists"